### PR TITLE
Nominal duration

### PR DESCRIPTION
**Pull request by ettomatic**
Change recipes definition to set timings in `s` and `ms`.

An example:

```JSON
{
    "origin": "https://www.bbc.co.uk/news",
    "stages": [
        { "at": 0,    "status": 404, "latency": "50ms"},
        { "at": "4s", "status": 503, "latency": "2s"},
        { "at": "9s", "status": 200, "latency": "100ms..200ms"}
    ]
}
```

### CODE REVIEW

File: test/origin_simulator/duration_test.exs [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+    test "0s" do
+      assert Duration.parse("0s") == 0
+    end
+
+    test "100ms" do
+      assert Duration.parse("100ms") == 100
+    end
+
+    test "1ms" do
```
Comment (samfrench) [inline test/origin_simulator/duration_test.exs]: This looks to be 1s.

File: test/origin_simulator/duration_test.exs [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+      assert Duration.parse("1s..2s") == 1000..2000
+    end
+
+    test "range 800ms..1s" do
+      assert Duration.parse("800ms..1s") == 800..1000
+    end
+  end
+
+  describe "parsing a tuple" do
+    test "0" do
```
Comment (samfrench) [inline test/origin_simulator/duration_test.exs]: ~0, "" or just 0 ?~

Actually, this is just 0 as the others have a time string.

File: lib/origin_simulator/duration.ex [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  def parse(0), do: 0
+
+  def parse(time) when is_integer(time) do
+    raise("Invalid stage latency, please define time in s or ms")
+  end
+
+  def parse(time) when is_binary(time) do
+    cond do
+      String.contains?(time, "..") -> String.split(time, "..")
+      true -> Integer.parse(time)
```
Comment (james-bowers) [inline lib/origin_simulator/duration.ex:11]: 👏  far better than bringing in a library for this

File: test/origin_simulator/duration_test.exs [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+    test "0s" do
+      assert Duration.parse({0, "s"}) == 0
+    end
+
+    test "100ms" do
+      assert Duration.parse({100, "ms"}) == 100
+    end
+
+    test "1ms" do
```
Comment (samfrench) [inline test/origin_simulator/duration_test.exs]: Again this should be s.

File: lib/origin_simulator/duration.ex (hunk 3 above)
Comment (ettomatic) [reply lib/origin_simulator/duration.ex:11]: `Integer.parse` is a nice little gem :) 
```elixir
iex(1)> Integer.parse("123s")
{123, "s"}
```

File: lib/origin_simulator/duration.ex [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+  def parse({time, "ms"}), do: time
+
+  def parse({time, "s"}), do: time * 1000
+
+  def parse({0, _}), do: 0
+
+  def parse(times) when is_list(times) do
+    min_max = Enum.map(times, fn (t) -> parse(t) end)
+    apply(Range, :new, min_max)
```
Comment (james-bowers) [inline lib/origin_simulator/duration.ex]: Could we do this instead?
```
def parse([min, max]) do
  Range.new(parse(min), parse(max))
end
```

If not, I think we can shorten the map line to:
`Enum.map(times, &parse/1)`

File: test/origin_simulator/duration_test.exs (hunk 1 above)
Comment (ettomatic) [reply test/origin_simulator/duration_test.exs]: copy and paste....

File: test/origin_simulator/duration_test.exs (hunk 2 above)
Comment (ettomatic) [reply test/origin_simulator/duration_test.exs]: nice! I just discovered I can label a test with an Integer :D

File: lib/origin_simulator/duration.ex (hunk 5 above)
Comment (ettomatic) [reply lib/origin_simulator/duration.ex]: nice idea! this?
```elixir
  def parse([min, max])  do
    Range.new(parse(min), parse(max))
  end
```

File: test/origin_simulator/duration_test.exs [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+    test "0s" do
+      assert Duration.parse("0s") == 0
+    end
+
+    test "100ms" do
+      assert Duration.parse("100ms") == 100
+    end
+
+    test "1ms" do
+      assert Duration.parse("1s") == 1000
```
Comment (james-bowers) [inline test/origin_simulator/duration_test.exs:34]: can we add a test for what happens if this occurs: `Duration.parse("1.2s")`?

Comment (ettomatic) [reply test/origin_simulator/duration_test.exs:34]: done

File: test/origin_simulator/duration_test.exs [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+      assert Duration.parse("800ms..1s") == 800..1000
+    end
+
+    test "invalid range" do
+      assert_raise RuntimeError, "Invalid timing, please define time in s or ms", fn () -> Duration.parse("1..2s") end
+    end
+  end
+
+  describe "parsing a tuple" do
+    test 0 do
```
Comment (samfrench) [inline test/origin_simulator/duration_test.exs:55]: These don't have to be quoted.

**Review by samfrench — APPROVED** [review]

File: lib/origin_simulator/duration.ex (hunk 3 above)
Comment (JoeARO) [reply lib/origin_simulator/duration.ex:11]: ooo I didn't know about this!

**Review by JoeARO — APPROVED** [review]

**Pull request merged**